web-extension/background-engine.js

```diff
@@ -17,19 +55,28 @@ bmcMessaging.addWindowHandler(
     evData => evData.type === 'computation'
               && evData.module === 'sources'
               && evData.computation === 'URL:Generate:Request',
-    evData => { // Don't receive `sender`, as we don't use it.
+    (evData, sender, sendResponse) => { // Don't receive `sender`, as we don't use it.
         LOGS.log('S68', {'evData' :evData});
-        const answerEv = {
-            type: 'computation',
-            module: 'sources',
-            computation: 'URL:Generate:Response',
-            resource: {
-                reader: evData.resource.reader,
-                comic: evData.resource.comic,
-                url: bmcSources.computeURL(evData.resource.reader,
-                                           evData.resource.comic),
+        bmcData.getComic(evData.resource.id, (err, comic) => {
+            let answerEv = {
+                type: 'computation',
+                module: 'sources',
+                computation: 'URL:Generate:Response',
+                resource: {
+                    err: false,
+                    url: null,
+                },
+            };
+            if (err) {
+                answerEv.err = 'Failed to get Comic data from storage';
```

Comment:
I just comment on it but no need to pay attention: you didn't use translated string stuff. But considering that we want to re-write it, I guess it's fine...

Reply:
Nah, you're right, I'll make it fit the current model, I'd rather have something that we want to change and is consistent, than create inconsistency.